bool reset_counter
---
bool result
